dartagnan/src/main/java/com/dat3m/dartagnan/wmm/Wmm.java

```diff
@@ -143,10 +142,10 @@ public Relation addDefinition(Definition definition) {
     }
 
     public void removeDefinition(Relation definedRelation) {
-        checkArgument(!(definedRelation.definition instanceof Definition.Undefined),
-                "Already undefined relation %s.", definedRelation);
-        logger.trace("Remove definition {}", definedRelation.definition);
-        definedRelation.definition = new Definition.Undefined(definedRelation);
+        if (!(definedRelation.getDefinition() instanceof Definition.Undefined)) {
+            logger.trace("Remove definition {}", definedRelation.definition);
+            definedRelation.definition = new Definition.Undefined(definedRelation);
+        }
```

Comment:
For consistency, if you use `definedRelation.getDefinition()` in the first case, you should not use `definedRelation.definition` in the second case